### PR TITLE
Update with Tom's suggestions

### PR DESCRIPTION
**Pull request by maxhchen**
Handle invalid requests for species data; split species.js into pixelation.js and fetchData.js

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -39,7 +39,8 @@ <h1>Pieced Homepage</h1>
 
         <!-- Access to test JSON file; remove later -->
         <script src="test.json"></script>
-        <script src="species.js"></script>
+    <script src="scripts/pixelation.js"></script>
+    <script src="scripts/fetchData.js"></script>
```
Comment (fgoyal) [inline src/main/webapp/index.html:47]: I'm not sure if this is just github, but the indentation of these two don't match the rest

File: src/main/webapp/scripts/pixelation.js [hunk 2]
```diff
@@ -64,14 +63,14 @@ function animate_update(endpoint) {
 
         if (PIXEL_FACTOR_CURR < target) {
             PIXEL_FACTOR_CURR += dx;
-            // "Binary Search" appraoch to hone in on exact value since dx > 1
+            // "Binary Search" approach to home in on exact value since dx > 1
```
Comment (fgoyal) [inline src/main/webapp/scripts/pixelation.js:66]: I think this was actually was supposed to be 'hone'

**Review by fgoyal — APPROVED** [review]

File: src/main/webapp/index.html (hunk 1 above)
Comment (maxhchen) [reply src/main/webapp/index.html:47]: Oops missed that, thanks

File: src/main/webapp/scripts/pixelation.js (hunk 2 above)
Comment (maxhchen) [reply src/main/webapp/scripts/pixelation.js:66]: Yeah, but I think both work anyways

**Review by fgoyal — APPROVED** [review]

**Review by arianazhu — APPROVED** [review]

**Pull request merged**